aiida_quantumespresso/parsers/cp.py

```diff
@@ -39,140 +40,181 @@ def parse(self, **kwargs):
-                raw_trajectory['pressure'] = matrix[:, 9]  # out_press, GPa
-                raw_trajectory['evp_times'] = matrix[:, 10]  # TPS, ps
-
-            # Huristics to understand if it's correct.
-            # A better heuristics could also try to fix possible issues
-            # (in new versions of QE, it's possible to recompile it with
-            # the __OLD_FORMAT flag to get back the old version format...)
-            # but I won't do it, as there may be also other columns swapped.
-            # Better to stop and ask the user to check what's going on.
-            max_time_difference = abs(numpy.array(raw_trajectory['times']) -
-                                      numpy.array(raw_trajectory['evp_times'])).max()
-            if max_time_difference > 1.e-4:  # It is typically ~1.e-7 due to roundoff errors
-                # If there is a large discrepancy
-                # it means there is something very weird going on...
-                return self.exit(self.exit_codes.ERROR_READING_TRAJECTORY_DATA)
-
-            # Delete evp_times in any case, it's a duplicate of 'times'
-            del raw_trajectory['evp_times']
-        except IOError:
-            out_dict['warnings'].append('Unable to open the EVP file... skipping.')
-
-        # get the symbols from the input
-        # TODO: I should have kinds in TrajectoryData
-        input_structure = self.node.inputs.structure
-        raw_trajectory['symbols'] = [str(i.kind_name) for i in input_structure.sites]
-
-        traj = TrajectoryData()
-        traj.set_trajectory(
-            stepids=raw_trajectory['steps'],
-            cells=raw_trajectory['cells'],
-            symbols=raw_trajectory['symbols'],
-            positions=raw_trajectory['positions_ordered'],
-            times=raw_trajectory['times'],
-            velocities=raw_trajectory['velocities_ordered'],
-        )
+            # get the symbols from the input
+            # TODO: I should have kinds in TrajectoryData
+            input_structure = self.node.inputs.structure
+            raw_trajectory['symbols'] = [str(i.kind_name) for i in input_structure.sites]
+
+            traj = TrajectoryData()
+            traj.set_trajectory(
+                stepids=raw_trajectory['steps'],
+                cells=raw_trajectory['cells'],
+                symbols=raw_trajectory['symbols'],
+                positions=raw_trajectory['positions_ordered'],
+                times=raw_trajectory['times'],
+                velocities=raw_trajectory['velocities_ordered'],
+            )
 
-        for this_name in evp_keys:
+            # eventually set the forces
             try:
-                traj.set_array(this_name, raw_trajectory[this_name])
-            except KeyError:
-                # Some columns may have not been parsed, skip
-                pass
+                traj.set_array('forces', raw_trajectory['forces_ordered'])
+            except:
```

Comment:
Let's use a more specific exception (KeyError?). Moreover, could you check what happens if the file is empty?